dataflux_core/download.py

```diff
@@ -156,6 +160,50 @@ def __init__(self, max_composite_object_size):
         self.max_composite_object_size = max_composite_object_size
 
 
+def dataflux_download_parallel(
+    project_name: str,
+    bucket_name: str,
+    objects: list[tuple[str, int]],
+    storage_client: object = None,
+    dataflux_download_optimization_params: DataFluxDownloadOptimizationParams = None,
+    parallelization: int = 1,
+) -> list[bytes]:
+    """Perform the DataFlux download algorithm in parallel to download the object contents as bytes and return.
+
+    Args:
+        project_name: the name of the GCP project.
+        bucket_name: the name of the GCS bucket that holds the objects to compose.
+            The function uploads the the composed object to this bucket too.
+        objects: A list of tuples which indicate the object names and sizes (in bytes) in the bucket.
+            Example: [("object_name_A", 1000), ("object_name_B", 2000)]
+        storage_client: the google.cloud.storage.Client initialized with the project.
+            If not defined, the function will initialize the client with the project_name.
+        dataflux_download_optimization_params: the paramemters used to optimize the download performance.
+        parallelization: The number of parallel processes that will simultaneously execute the download.
+    Returns:
+        the contents of the object in bytes.
+    """
+    chunk_size = math.ceil(len(objects) / parallelization)
```

Comment:
Won't this silently drop the last batch sometimes?

if __name__ == "__main__":
    objects = [1, 2, 3, 4, 5]
    parallelization = 3
    chunk_size = math.ceil(len(objects) / parallelization)
    chunks = [objects[i * chunk_size : (i + 1) * chunk_size] for i in range(2)]
    print(chunks)

this returns [[1, 2], [3, 4]]

Reply:
If the paralellization was set to. 3, it would be `for i in range(3)` so it would it would still pick up all objects with item 5 being stored in an array of its own. I'm confident the code won't drop items at the end.

Reply:
oh sorry I input the wrong stuff. I meant to say this

if __name__ == "__main__":
    objects = [1, 2, 3, 4, 5]
    parallelization = 4
    chunk_size = math.ceil(len(objects) / parallelization)
    chunks = [
        objects[i * chunk_size : (i + 1) * chunk_size] for i in range(parallelization)
    ]
    print(chunks)

this gives [[1, 2], [3, 4], [5], []]

Reply:
Yeah that's totally fine, doesn't cause issues. I even included unit tests for exceeding max object size.

Reply:
no I meant the empty list returned there. Can we avoid creating a process for the empty list?